File: bubbling.rb


def switch(array, index)
	i = 0
	new_array =[]
	while i < array.length
		if i == index[0]
			new_array << array[index[1]]
		elsif i == index[1]
			new_array << array[index[0]]
		else
			new_array << array[i]
		end

		i +=1	
	end
	return new_array
end

# puts switch(["a", "b", "c", "d","e", "f"], [3, 4])



def bubbleing(array)
	no_switch = 0
	while no_switch < ((array.length) -1)
		i = 0
		j = 1
		while j < array.length
			if array[i] > array[j]
				array = switch(array, [i, j])

			else
				no_switch = no_switch + 1
			end
			i +=1
			j +=1
		end
		if no_switch != ((array.length) -1)
			no_switch = 0 
		end
	end
	return array
end

puts bubbleing([2, 11, 10, 4, 1])

File: bubbling.js
function switch_up(array, index){
	var new_array = new Array();
	
	for(i = 0; i < array.length; i++){
		if (i == index[0]){
			new_array.push(array[index[1]]);
			
		}
		else if (i == index[1]){
			new_array.push(array[index[0]]);
			
		}
		else {
			new_array.push(array[i]);
			
		}

	}
	return new_array;
}

// console.log(switch_up([4, 3, 5, 6], [0,1]));



function bubbling(array){
	var no_switch = 0;
	while (no_switch < ((array.length)- 1)){
		var i = 0;
		var j = 1;
		while(j < array.length){
			if(array[i] > array[j]){
				array = switch_up(array,[i, j]);
			}
			else{
				no_switch = no_switch + 1;
			}
			i++
			j++
		}
		if(no_switch!= ((array.length) - 1)){
			no_switch = 0; 
		}
	}
	return array;
}

console.log(bubbling([11, 1, 4, 2]));